src/preprocessing/passes/bool_to_bv.cpp

```diff
@@ -394,33 +403,18 @@ void BoolToBV::rebuildNode(const TNode& n, Kind new_kind)
 }
 
 BoolToBV::Statistics::Statistics()
-    : d_numIteToBvite("preprocessing::passes::BoolToBV::NumIteToBvite", 0),
-      d_numTermsLowered("preprocessing::passes:BoolToBV::NumTermsLowered", 0),
-      d_numIntroducedItes(
-          "preprocessing::passes::BoolToBV::NumTermsForcedLowered", 0)
-{
-  smtStatisticsRegistry()->registerStat(&d_numIteToBvite);
-  if (options::boolToBitvector() == options::BoolToBVMode::ALL)
-  {
-    // these statistics wouldn't be correct in the ITE mode,
-    // because it might discard rebuilt nodes if it fails to
-    // convert a bool to width-one bit-vector (never forces)
-    smtStatisticsRegistry()->registerStat(&d_numTermsLowered);
-    smtStatisticsRegistry()->registerStat(&d_numIntroducedItes);
-  }
-}
-
-BoolToBV::Statistics::~Statistics()
+    : d_numIteToBvite(smtStatisticsRegistry().registerInt(
+        "preprocessing::passes::BoolToBV::NumIteToBvite")),
+      // the following two statistics are not correct in the ITE mode, because
+      // we might discard rebuilt nodes if we fails to convert a bool to
+      // width-one bit-vector (never forces)
+      d_numTermsLowered(smtStatisticsRegistry().registerInt(
```

Comment:
One point of discussion here: 
The alternative change here would only register the stats if the option is enabled.  Is it still possible to do this?  Then, I guess the calls to increment stats would have to be guarded on whether we have allocated the stat.

In this case, this is a pretty minor issue (I don't mind this minor change in behavior), but I do think we should (even aggressively) disable statistics that are not relevant.  This would make the `--stats` output contain less junk.

Reply:
This is somewhat difficult because the only way to create a statistic is by obtaining it from the registry.

However, I see no big problem "spamming" the statistics: both when printing and when querying in the API, only statistics that have actually been changed are shown. (There is `--stats-unset` to show all of them)

Reply:
Showing only the statistics that have changed is a change in default behavior, right?

Reply:
Yes, it is.

Reply:
I took a small QF_NRA example
- `--stats`: https://pastebin.com/LmbJvtkD
- `--stats --stats-expert`: https://pastebin.com/Gx1m8Fha
- `--stats --stats-unset`: https://pastebin.com/x51niMqG
- `--stats --stats-expert --stats-unset`: https://pastebin.com/jBTBqqdC

My intention would be that `--stats` is sufficient for most users, `--stats-expert` is what experienced users and developers look at, and `--stats-unset` is only used when you are working on statistics themselves.

Reply:
Discussed offline, changed `--stats-unset` to `--stats-all` and decided to change behavior slightly.